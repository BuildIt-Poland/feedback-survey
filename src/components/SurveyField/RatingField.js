import React from 'react';
import PropTypes from 'prop-types';

import { formField, radioOption } from '../../types';
import CustomRadioButtonGroup from '../CustomRadioButtonGroup';
import StyledFormField from './StyledFormField';

const RatingField = ({ field, options }) => (
  <StyledFormField
    name={field.id}
    label={field.content}
    required={field.required}
    component={CustomRadioButtonGroup}
    options={options}
    direction="row-responsive"
    width="48rem"
    withIcon={true}
  />
);

RatingField.propTypes = {
  field: formField.isRequired,
  options: PropTypes.arrayOf(radioOption)
};

export default RatingField;
