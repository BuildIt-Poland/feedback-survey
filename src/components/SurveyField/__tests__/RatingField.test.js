import React from 'react';
import { create } from 'react-test-renderer';

import RatingField from '../RatingField';

describe('COMPONENT - SurveyField RatingField', () => {
  it('does not render without field', () => {
    const field = {
      id: '123',
      type: 'rating',
      content: 'Question content',
      required: false,
      options: ['Bad', 'Average']
    };
    const component = create(<RatingField field={field} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
