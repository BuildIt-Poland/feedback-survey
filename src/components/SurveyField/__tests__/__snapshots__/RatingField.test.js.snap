// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - SurveyField RatingField does not render without field 1`] = `
<div
  className="sc-ifAKCX ittBkw FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
>
  <label
    className="StyledText-sc-1sadyjn-0 bMXxti"
  >
    Question content
  </label>
  <div
    className="StyledBox-sc-13pk1d4-0 gZSidW"
  >
    <div
      className="sc-bxivhb iPHzoF"
    >
      <input
        className="sc-bdVaJa kRURFE"
        id="test123"
        name="fieldName"
        type="radio"
        value="test"
      />
      <label
        className="sc-bwzfXH cxGQZ"
        htmlFor="test123"
      >
        <svg
          viewBox="0 0 490 490"
          x="0px"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <g>
            <circle
              cx="157.651"
              cy="175.288"
              r="42.642"
            />
            <circle
              cx="332.349"
              cy="175.289"
              r="42.642"
            />
            <path
              d="M69.086,490h351.829C459.001,490,490,459.001,490,420.914V69.086C490,30.999,459.001,0,420.914,0H69.086 C30.991,0,0,30.999,0,69.086v351.829C0,459.001,30.991,490,69.086,490z M30.625,69.086c0-21.204,17.257-38.461,38.461-38.461 h351.829c21.204,0,38.461,17.257,38.461,38.461v351.829c0,21.204-17.257,38.461-38.461,38.461H69.086 c-21.204,0-38.461-17.257-38.461-38.461V69.086z"
            />
          </g>
        </svg>
        <span>
          label-test
        </span>
      </label>
    </div>
  </div>
</div>
`;
